pkg/controller/common.go

```diff
@@ -284,7 +286,9 @@ func (c *SharedState) createOperationQueue(volumeName string) {
 	_, _ = c.volumeOperationQueues.LoadOrStore(volumeName, newLockableEntry(newOperationQueue()))
 }
 
-func (c *SharedState) addToOperationQueue(volumeName string, requester operationRequester, operationFunc func() error, isReplicaGarbageCollection bool) {
+func (c *SharedState) addToOperationQueue(ctx context.Context, volumeName string, requester operationRequester, operationFunc func(context.Context) error, isReplicaGarbageCollection bool) {
+	ctx, w := workflow.New(ctx, workflow.WithDetails(consts.VolumeNameLabel, volumeName))
```

Comment:
While it is terribly convenient and reasonable to create the workflow here, `addToOperationQueue` will be recorded as the initiator rather than the caller (i.e. `createReplicas` and `triggerManageReplicas`) which isn't very information. Consider adding a `workflow.WithCaller(skip int)` option that would allow us to capture a caller explicitly here, e.g.:

```go
ctx, w := workflow.New(cts, workflow.WithCaller(1), workflow.WithDetails(consts.VolumeNameLabel, volumeName))
```

Reply:
Oh, I see now that the callers both create parent workflows and this simply creates a child workflow for the queueing operation. Setting the caller explicitly here wouldn't make sense in that case.